src/test/java/org/dice_research/opal/launuts/polygons/parser/NutParserTest.java

```diff
@@ -0,0 +1,169 @@
+package org.dice_research.opal.launuts.polygons.parser;
+
+import java.io.FileNotFoundException;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.dice_research.opal.launuts.polygons.MultiPolygon;
+import org.dice_research.opal.launuts.polygons.PolygonParserException;
+import org.dice_research.opal.launuts.polygons.parser.NutsParser;
+import org.junit.Test;
+
+/**
+ * In this test class, we will perform two tests.
+ * 
+ * The first test is to confirm the integrity of holes inside nuts in
+ * NUTS_Polygons.json file.
+ * 
+ * The second test is to confirm the validity of a few important methods which
+ * were used in NutsParser class to extract polygons from GeoJson files provides
+ * from Eurostat.
+ * 
+ * @author Gourab Sahu
+ *
+ */
+
+public class NutParserTest {
+
+	public TestMultiPolygons tester = new TestMultiPolygons();
+	public NutsParser nut_parser = new NutsParser();
+	MultiPolygon hole;
+
+	// Pforzheim_Stadtkreis is a hole inside pforzheim_stadtkreis
+	private static String nuts_code_of_Enzkreis = "DE12B";
+	private static String nuts_code_of_Pforzheim_Stadtkreis = "DE129";
+
+	// Eisenach_Kreisfreie_Stadt is a hole inside Wartburgkreis
+	private static String nuts_code_of_Wartburgkreis = "DEG0P";
+	private static String nuts_code_of_Eisenach_Kreisfreie_Stadt = "DEG0N";
+
+	// Weimar Kreisfreie_Stadt is a hole inside Weimarer Land
+	private static String nuts_code_of_Weimarer_Land = "DEG0G";
+	private static String nuts_code_of_Weimar_Kreisfreie_Stadt = "DEG05";
+
+	// Landau_in_der_Pfalz_Kreisfreie_Stadt is a hole inside Südliche_Weinstraße
+	private static String nuts_code_of_Südliche_Weinstraße = "DEB3H";
+	private static String nuts_code_of_Landau_in_der_Pfalz_Kreisfreie_Stadt = "DEB33";
+
+	// Baden_baden is a hole inside Rastaat
+	private static String nuts_code_of_Rastaat = "DEB3H";
+	private static String nuts_code_of_Baden_Baden = "DEB33";
+
+	// Berlin is a hole inside Brandenburg
+	private static String nuts_code_of_Brandenburg = "DE40";
+	private static String nuts_code_of_Berlin = "DE3";
+
+	// Bremen is a hole inside Niedersachsen
+	private static String nuts_code_of_Niedersachsen = "DE9";
+	private static String nuts_code_of_Bremen = "DE501";
+
+	// Ansbach is a hole inside Ansbach Landkreis
+	private static String nuts_code_of_Ansbach_Landkreis = "DE256";
+	private static String nuts_code_of_Ansbach = "DE251";
+
+	// Ansbach is a hole inside Ansbach Landkreis
+	private static String nuts_code_of_Coburg_Landkreis = "DE247";
+	private static String nuts_code_of_Coburg_Kreisfreie_Stadt = "DE243";
```

Comment:
Initialised a lot of unnecessary variables.
It is really not a good coding practice, if you just need them to use for your test cases and not need to reuse them. See, if you you can just pass the values directly in your test functions.

Reply:
Those are supposed to be Final static. They represent a particular a hole and a nuts where hole exists. I will change them to constants.